### PR TITLE
Slightly delay inline tags so that they don't immediately appear when user it typing a longer shortcut.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
A couple of people have mentioned that inline-tags appear as they're typing `ctrl-alt-<some-other-char>`.  For example `ctrl-alt-p` is a debugger shortcut.  This can be distracting/annoying/flashing.

I've tweaked things to put inline-hints on a slightly longer delay.  Manual testing here found adding a 100ms delay here struck a sweet spot of having these no longer appear when typing another shortcut, while still feeling like they come up quickly when you do want them.  

If this  isn't a suitable fix in practice, we can consider a different trigger chord that either isn't the prefix of other commands, or is only a prefix for a suitably tiny set of rare commands such that users will be very unlikely to see this.

This PR sholud be reviewed commit by commit.  The first commit is a rename for clarity.  The second introduces the new behavior.

### CODE REVIEW

**Review by NTaylorMullen — APPROVED** [review]

**Comment by sharwell** [comment]
This is going to make the interference with multiple carets even worse ☹️

File: src/EditorFeatures/Core/InlineHints/InlineHintsDataTaggerProvider.cs [hunk 1]
```diff
@@ -61,7 +61,10 @@ protected override ITaggerEventSource CreateEventSource(ITextView textViewOpt, I
             return TaggerEventSources.Compose(
                 TaggerEventSources.OnViewSpanChanged(ThreadingContext, textViewOpt, textChangeDelay: TaggerDelay.Short, scrollChangeDelay: TaggerDelay.Immediate),
                 TaggerEventSources.OnWorkspaceChanged(subjectBuffer, TaggerDelay.Immediate, _listener),
-                TaggerEventSources.OnOptionChanged(subjectBuffer, InlineHintsOptions.DisplayAllOverride, TaggerDelay.Immediate),
+                // Ctrl-alt is the initial trigger for a lot of features (like ctrl-alt-p, attach to debugger). So bring
+                // this up not quite immediately so that users don't get flashes of this appearing when running other
+                // commands.
+                TaggerEventSources.OnOptionChanged(subjectBuffer, InlineHintsOptions.DisplayAllOverride, TaggerDelay.NearImmediate),
```
Comment (sharwell) [inline src/EditorFeatures/Core/InlineHints/InlineHintsDataTaggerProvider.cs:67]: 💭 We really need this to start calculation on `Immediate` delay, but avoid rendering notification faster than `NearImmediate`. I've seen cases where calculation is not immediate and this delay can be used for latency hiding.

**Review by sharwell — CHANGES_REQUESTED** [review]
This change assumes that <kbd>Ctrl</kbd>+<kbd>Alt</kbd> is the trigger for this feature. However, this is already bound to multi-caret insertion, which is heavily mouse driven and cannot be impaired by a feature that moves characters around on the screen. Finding a new keyboard trigger for inline hints is required, and only after we identify the new shortcut will we know what the possible other interactions are.

**Comment by CyrusNajmabadi** [comment]
Good point.  @olegtk @AmadeusW do you know if shift-space is used for anything?

**Comment by NTaylorMullen** [comment]
> do you know if shift-space is used for anything?

Technically it enters a "space" today into the editor; however, if we were to change that I'd be psyched 😄

**Comment by sharwell** [comment]
> Technically it enters a "space" today into the editor; however, if we were to change that I'd be psyched 😄

I'm going to guess a bunch of people use this to TYPE CAPITAL LETTERS IN COMMENTS.

**Comment by NTaylorMullen** [comment]
> TYPE CAPITAL LETTERS IN COMMENTS.

BUT BUT BUT I WANT IT

**Comment by CyrusNajmabadi** [comment]
> Technically it enters a "space" today into the editor; however, if we were to change that I'd be psyched 😄

It wouldn't change anything.  We never capture these keys.  We just use the signal that they're held down (for some amount of time) to trigger this.  So the user would have to be holding shift-spaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaace to have an issue here.  THat said, someone might be doing that... i.e. `NOW                 IS                THE               TIME`.  Not sure if that's an issue or not.

**Comment by CyrusNajmabadi** [comment]
Nevermind.  shift-space is not usable at all in any evnet.  Because if someone just wants to see the inline hint, it would never be ok for that to insert space.

**Comment by CyrusNajmabadi** [comment]
Closing out.  We sidestepped this by changing the keybinding to Alt-F1

**Pull request closed without merging**